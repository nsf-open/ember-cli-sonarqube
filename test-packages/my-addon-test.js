const { sonar, readFile } = require('./test-helpers');

describe('for a fully configured addon', function () {
  jest.setTimeout(10000000);

  it('it generates files for Sonarqube', async function () {
    await expect(
      sonar('my-addon', ['--test-cmd="npx ember test"'])
    ).resolves.not.toThrow();

    expect(readFile('my-addon', 'coverage/eslint-out.json')).not.toBeFalsy();
    expect(readFile('my-addon', 'coverage/template-lint-out.json')).not.toBeFalsy();
    expect(readFile('my-addon', 'coverage/lcov.info')).not.toBeFalsy();
    expect(readFile('my-addon', 'coverage/test-report.xml')).not.toBeFalsy();
  });
});
