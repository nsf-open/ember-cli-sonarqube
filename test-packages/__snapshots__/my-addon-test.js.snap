// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`for a fully configured addon it generates files for Sonarqube 1`] = `"[{\\"filePath\\":\\"/Users/mdjones/Documents/NSF/Github/nsf-open/ember-cli-sonarqube/test-packages/my-addon/.eslintrc.js\\",\\"messages\\":[],\\"errorCount\\":0,\\"fatalErrorCount\\":0,\\"warningCount\\":0,\\"fixableErrorCount\\":0,\\"fixableWarningCount\\":0,\\"usedDeprecatedRules\\":[{\\"ruleId\\":\\"no-process-exit\\",\\"replacedBy\\":[]}]},{\\"filePath\\":\\"/Users/mdjones/Documents/NSF/Github/nsf-open/ember-cli-sonarqube/test-packages/my-addon/.prettierrc.js\\",\\"messages\\":[],\\"errorCount\\":0,\\"fatalErrorCount\\":0,\\"warningCount\\":0,\\"fixableErrorCount\\":0,\\"fixableWarningCount\\":0,\\"usedDeprecatedRules\\":[{\\"ruleId\\":\\"no-process-exit\\",\\"replacedBy\\":[]}]},{\\"filePath\\":\\"/Users/mdjones/Documents/NSF/Github/nsf-open/ember-cli-sonarqube/test-packages/my-addon/.template-lintrc.js\\",\\"messages\\":[],\\"errorCount\\":0,\\"fatalErrorCount\\":0,\\"warningCount\\":0,\\"fixableErrorCount\\":0,\\"fixableWarningCount\\":0,\\"usedDeprecatedRules\\":[{\\"ruleId\\":\\"no-process-exit\\",\\"replacedBy\\":[]}]},{\\"filePath\\":\\"/Users/mdjones/Documents/NSF/Github/nsf-open/ember-cli-sonarqube/test-packages/my-addon/addon/index.js\\",\\"messages\\":[],\\"errorCount\\":0,\\"fatalErrorCount\\":0,\\"warningCount\\":0,\\"fixableErrorCount\\":0,\\"fixableWarningCount\\":0,\\"usedDeprecatedRules\\":[]},{\\"filePath\\":\\"/Users/mdjones/Documents/NSF/Github/nsf-open/ember-cli-sonarqube/test-packages/my-addon/config/ember-try.js\\",\\"messages\\":[],\\"errorCount\\":0,\\"fatalErrorCount\\":0,\\"warningCount\\":0,\\"fixableErrorCount\\":0,\\"fixableWarningCount\\":0,\\"usedDeprecatedRules\\":[{\\"ruleId\\":\\"no-process-exit\\",\\"replacedBy\\":[]}]},{\\"filePath\\":\\"/Users/mdjones/Documents/NSF/Github/nsf-open/ember-cli-sonarqube/test-packages/my-addon/config/environment.js\\",\\"messages\\":[],\\"errorCount\\":0,\\"fatalErrorCount\\":0,\\"warningCount\\":0,\\"fixableErrorCount\\":0,\\"fixableWarningCount\\":0,\\"usedDeprecatedRules\\":[{\\"ruleId\\":\\"no-process-exit\\",\\"replacedBy\\":[]}]},{\\"filePath\\":\\"/Users/mdjones/Documents/NSF/Github/nsf-open/ember-cli-sonarqube/test-packages/my-addon/ember-cli-build.js\\",\\"messages\\":[],\\"errorCount\\":0,\\"fatalErrorCount\\":0,\\"warningCount\\":0,\\"fixableErrorCount\\":0,\\"fixableWarningCount\\":0,\\"usedDeprecatedRules\\":[{\\"ruleId\\":\\"no-process-exit\\",\\"replacedBy\\":[]}]},{\\"filePath\\":\\"/Users/mdjones/Documents/NSF/Github/nsf-open/ember-cli-sonarqube/test-packages/my-addon/index.js\\",\\"messages\\":[],\\"errorCount\\":0,\\"fatalErrorCount\\":0,\\"warningCount\\":0,\\"fixableErrorCount\\":0,\\"fixableWarningCount\\":0,\\"usedDeprecatedRules\\":[{\\"ruleId\\":\\"no-process-exit\\",\\"replacedBy\\":[]}]},{\\"filePath\\":\\"/Users/mdjones/Documents/NSF/Github/nsf-open/ember-cli-sonarqube/test-packages/my-addon/testem.js\\",\\"messages\\":[],\\"errorCount\\":0,\\"fatalErrorCount\\":0,\\"warningCount\\":0,\\"fixableErrorCount\\":0,\\"fixableWarningCount\\":0,\\"usedDeprecatedRules\\":[{\\"ruleId\\":\\"no-process-exit\\",\\"replacedBy\\":[]}]},{\\"filePath\\":\\"/Users/mdjones/Documents/NSF/Github/nsf-open/ember-cli-sonarqube/test-packages/my-addon/tests/acceptance/index-test.js\\",\\"messages\\":[],\\"errorCount\\":0,\\"fatalErrorCount\\":0,\\"warningCount\\":0,\\"fixableErrorCount\\":0,\\"fixableWarningCount\\":0,\\"usedDeprecatedRules\\":[]},{\\"filePath\\":\\"/Users/mdjones/Documents/NSF/Github/nsf-open/ember-cli-sonarqube/test-packages/my-addon/tests/dummy/app/app.js\\",\\"messages\\":[],\\"errorCount\\":0,\\"fatalErrorCount\\":0,\\"warningCount\\":0,\\"fixableErrorCount\\":0,\\"fixableWarningCount\\":0,\\"usedDeprecatedRules\\":[]},{\\"filePath\\":\\"/Users/mdjones/Documents/NSF/Github/nsf-open/ember-cli-sonarqube/test-packages/my-addon/tests/dummy/app/router.js\\",\\"messages\\":[],\\"errorCount\\":0,\\"fatalErrorCount\\":0,\\"warningCount\\":0,\\"fixableErrorCount\\":0,\\"fixableWarningCount\\":0,\\"usedDeprecatedRules\\":[]},{\\"filePath\\":\\"/Users/mdjones/Documents/NSF/Github/nsf-open/ember-cli-sonarqube/test-packages/my-addon/tests/dummy/config/environment.js\\",\\"messages\\":[],\\"errorCount\\":0,\\"fatalErrorCount\\":0,\\"warningCount\\":0,\\"fixableErrorCount\\":0,\\"fixableWarningCount\\":0,\\"usedDeprecatedRules\\":[{\\"ruleId\\":\\"no-process-exit\\",\\"replacedBy\\":[]}]},{\\"filePath\\":\\"/Users/mdjones/Documents/NSF/Github/nsf-open/ember-cli-sonarqube/test-packages/my-addon/tests/dummy/config/targets.js\\",\\"messages\\":[],\\"errorCount\\":0,\\"fatalErrorCount\\":0,\\"warningCount\\":0,\\"fixableErrorCount\\":0,\\"fixableWarningCount\\":0,\\"usedDeprecatedRules\\":[{\\"ruleId\\":\\"no-process-exit\\",\\"replacedBy\\":[]}]},{\\"filePath\\":\\"/Users/mdjones/Documents/NSF/Github/nsf-open/ember-cli-sonarqube/test-packages/my-addon/tests/helpers/index.js\\",\\"messages\\":[],\\"errorCount\\":0,\\"fatalErrorCount\\":0,\\"warningCount\\":0,\\"fixableErrorCount\\":0,\\"fixableWarningCount\\":0,\\"usedDeprecatedRules\\":[]},{\\"filePath\\":\\"/Users/mdjones/Documents/NSF/Github/nsf-open/ember-cli-sonarqube/test-packages/my-addon/tests/test-helper.js\\",\\"messages\\":[{\\"ruleId\\":\\"prettier/prettier\\",\\"severity\\":2,\\"message\\":\\"Delete \`‚èé\`\\",\\"line\\":13,\\"column\\":1,\\"nodeType\\":null,\\"messageId\\":\\"delete\\",\\"endLine\\":14,\\"endColumn\\":1,\\"fix\\":{\\"range\\":[423,424],\\"text\\":\\"\\"}}],\\"errorCount\\":1,\\"fatalErrorCount\\":0,\\"warningCount\\":0,\\"fixableErrorCount\\":1,\\"fixableWarningCount\\":0,\\"source\\":\\"import Application from 'dummy/app';\\\\nimport config from 'dummy/config/environment';\\\\nimport * as QUnit from 'qunit';\\\\nimport { setApplication } from '@ember/test-helpers';\\\\nimport { setup } from 'qunit-dom';\\\\nimport { start } from 'ember-qunit';\\\\nimport {\\\\n  forceModulesToBeLoaded,\\\\n  sendCoverage,\\\\n} from 'ember-cli-code-coverage/test-support';\\\\nimport { setupQunitReporting } from '@nsf-open/ember-cli-sonarqube/test-support';\\\\n\\\\n\\\\nsetApplication(Application.create(config.APP));\\\\n\\\\nQUnit.done(async function () {\\\\n  forceModulesToBeLoaded();\\\\n  await sendCoverage();\\\\n});\\\\n\\\\nsetupQunitReporting(QUnit);\\\\nsetup(QUnit.assert);\\\\n\\\\nstart();\\\\n\\",\\"usedDeprecatedRules\\":[]}]"`;

exports[`for a fully configured addon it generates files for Sonarqube 2`] = `"[{\\"filePath\\":\\"tests/dummy/app/templates/application.hbs\\",\\"messages\\":[{\\"ruleId\\":\\"no-curly-component-invocation\\",\\"severity\\":2,\\"message\\":\\"You are using the component {{moo}} with curly component syntax. You should use <Moo> instead. If it is actually a helper you must manually add it to the 'no-curly-component-invocation' rule configuration, e.g. \`'no-curly-component-invocation': { allow: ['moo'] }\`.\\",\\"line\\":5,\\"column\\":16},{\\"ruleId\\":\\"no-implicit-this\\",\\"severity\\":2,\\"message\\":\\"Ambiguous path 'moo' is not allowed. Use '@moo' if it is a named argument or 'this.moo' if it is a property on 'this'. If it is a helper or component that has no arguments, you must either convert it to an angle bracket invocation or manually add it to the 'no-implicit-this' rule configuration, e.g. 'no-implicit-this': { allow: ['moo'] }.\\",\\"line\\":5,\\"column\\":18}],\\"errorCount\\":2,\\"warningCount\\":0}]"`;

exports[`for a fully configured addon it generates files for Sonarqube 3`] = `
"TN:
SF:addon/index.js
FN:1,sum
FNF:1
FNH:0
FNDA:0,sum
DA:2,0
LF:1
LH:0
BRF:0
BRH:0
end_of_record
"
`;
